apps/create_charuco_boards/src/create_charuco_boards.cpp

```diff
@@ -59,27 +59,28 @@ int main(int argc, char *argv[]) {
   }
 
   // Create the charuco
-  cv::Ptr<cv::aruco::Dictionary> dict =
-      cv::aruco::getPredefinedDictionary(cv::aruco::DICT_6X6_1000);
-  std::vector<cv::Ptr<cv::aruco::CharucoBoard>> charucoBoards;
+  const cv::aruco::Dictionary dict = cv::aruco::getPredefinedDictionary(cv::aruco::DICT_6X6_250);
+  std::vector<cv::aruco::CharucoBoard> charucoBoards;
   int offset_count = 0;
   for (std::size_t i = 0; i < num_board; i++) {
     // declare the board
-    cv::Ptr<cv::aruco::CharucoBoard> charuco = cv::aruco::CharucoBoard::create(
-        number_x_square_per_board[i], number_y_square_per_board[i],
+    const cv::aruco::CharucoBoard charuco = cv::aruco::CharucoBoard(
+        cv::Size(number_x_square_per_board[i], number_y_square_per_board[i]),
         length_square, length_marker, dict);
+
     // If it is the first board then just use the standard idx
     if (i != 0) {
-      int id_offset = charucoBoards[i - 1]->ids.size() + offset_count;
+      int id_offset = charucoBoards[i - 1].getIds().size() + offset_count;
       offset_count = id_offset;
-      for (auto &id : charuco->ids) {
-        id += id_offset;
-      }
+      // for (auto &id : charuco->getIds()) {
+      //   id += id_offset;
+      // }
     }
```

Comment:
do you remember why was this needed at that time, @rameau-fr? It looks like some manipulation with indices of the first board. However, In the current version, the index vector is const so cannot be modified.

Reply:
This part also confused me the first time I read it. I believe the intention is to support multiple boards. In interpolateCornersCharuco (or CharucoDetector::detectBoard in OpenCV 4.10.0), it checks if the detected markerIds are present in the CharucoBoard's IDs. In OpenCV 4.10.0, the implementation of CharucoBoard is hidden using the Pimpl idiom, but we can still specify the IDs in its constructor.

Reply:
@BobbleLaw, thank you for the explanation. Fixed in a33577ce13e7165e54536689600b0f38d8bbd8ac

Reply:
It's okay. I refactored MC-Calib a lot to use in my project, so gonna keep my eyes on the changes from original authors (which are you guys), in case I misunderstand anything :rofl:.